### PR TITLE
OLS-517: Handle encoded/image data in index

### PR DESCRIPTION
**Pull request by syedriko**
- Added a timeout to the docs URL liveness check
- Added detection of bad nodes: base64 chunks.
  These are marked with `bad_node = 1` metadata label and are listed in the metadata.json file
- Added support for node lookup by node id

### CODE REVIEW

File: scripts/generate_embeddings.py [hunk 1]
```diff
@@ -107,8 +120,16 @@ def file_metadata_func(file_path: str) -> Dict:
...
+    bad_nodes = []
+    nodes = Settings.text_splitter.get_nodes_from_documents(documents)
+    for node in nodes:
+        if not got_whitespace(node.text):
+            bad_nodes.append(node.node_id)
+            node.metadata["bad_node"] = 1
+            print(f"bad node: {node.node_id}")
+
+    index = VectorStoreIndex(
+        nodes,
```
[the remark this answers is not in the document]
Comment (syedriko) [reply scripts/generate_embeddings.py:132]: That is a good point, I'll look into it. When I talked about being unable to delete individual nodes, I was working with a different implementation, one that traversed a vector database, looking for "bad" nodes.

[the remark this answers is not in the document]
Comment (syedriko) [reply scripts/generate_embeddings.py]: @bparees PTAL

**Comment by bparees** [comment]
@syedriko between this and your other update to filter out the OKE docs, i think it would be good to bump the OLS rag image ref after this merges+publishes

**Comment by syedriko** [comment]
> @syedriko between this and your other update to filter out the OKE docs, i think it would be good to bump the OLS rag image ref after this merges+publishes

+1

**Pull request merged**